presto-main/src/main/java/com/facebook/presto/execution/buffer/BroadcastOutputBuffer.java

```diff
@@ -310,6 +322,30 @@ public void fail()
         }
     }
 
+    @Override
+    public void setNoMorePagesForLifespan(Lifespan lifespan)
+    {
+        noMorePagesForLifespan.add(lifespan);
+    }
+
+    @Override
+    public void registerLifespanCompletionCallback(Consumer<Lifespan> callback)
+    {
+        checkState(lifespanCompletionCallback == null);
```

Comment:
`checkState(lifespanCompletionCallback == null, "lifespanCompletionCallback is already set");`